firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/ArrayTransformOperation.java

```diff
@@ -51,11 +51,11 @@ public FieldValue applyToRemoteDocument(FieldValue previousValue, FieldValue tra
   }
 
   @Override
-  public boolean equals(Object o) {
+  public final boolean equals(Object o) {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof ArrayTransformOperation)) {
```

Comment:
Hrm.  I'm not sure about this...  We have two implementations of ArrayTransformOperation: ArrayTransformOperation.Union, and ArrayTransformOperation.Remove.  They shouldn't be considered equal even if they have the same elements being unioned or removed.

I'm not sure what this means.  Honestly, getClass() equality is exactly what we want, and I'm not sure how much I care about the substitutability principle...  So we could keep the existing code but suppress the warning...  or we could override equals again in the Union / Remove classes to do the right thing.  WDYT?